trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/testing/ManagedS3MockContainer.java

```diff
@@ -28,24 +33,58 @@
 
 public class ManagedS3MockContainer
 {
-    private final S3MockContainer container;
+    private final MinIOContainer container;
+
+    private static final String CONFIG = """
+            {
+                "version": "10",
+                "aliases": {
+                    "local": {
+                        "url": "http://localhost:9000",
+                        "accessKey": "%s",
+                        "secretKey": "%s",
+                        "api": "S3v4",
+                        "path": "auto"
+                    }
+                }
+            }
+            """;
 
     @Retention(RUNTIME)
     @Target({FIELD, PARAMETER, METHOD})
     @BindingAnnotation
     public @interface ForS3MockContainer {}
 
     @Inject
-    public ManagedS3MockContainer(@ForS3MockContainer String initialBuckets)
+    public ManagedS3MockContainer(@ForS3MockContainer String initialBuckets, @ForS3MockContainer Credential credential)
     {
-        container = new S3MockContainer("3.8.0");
+        Transferable transferable = Transferable.of(CONFIG.formatted(credential.accessKey(), credential.secretKey()));
+        container = new MinIOContainer("minio/minio:RELEASE.2023-09-04T19-57-37Z")
+                .withUserName(credential.accessKey())
+                .withPassword(credential.secretKey())
+                // setting this allows us to shell into the container and run "mc" commands
+                .withCopyToContainer(transferable, "/root/.mc/config.json");
+
+        container.configure();
+        container.start();
+
         if (!initialBuckets.isEmpty()) {
-            container.withInitialBuckets(initialBuckets);
+            Splitter.on(',').trimResults().splitToStream(initialBuckets).forEach(bucket -> {
+                ExecResult execResult;
+                try {
+                    execResult = container.execInContainer("mc", "mb", "local/" + bucket);
```

Comment:
Could we use the `Minio` client instead? Similar to what `trino-testing-containers` does https://github.com/trinodb/trino/blob/d8f7efa41e98966355ffdfb0adfbb923b5a70ec0/testing/trino-testing-containers/src/main/java/io/trino/testing/containers/Minio.java#L115-L124

That way we don't need to mount that CONFIG object, and it also feels cleaner than invoking a cli
Wdyt?

Reply:
Mounting the CONFIG object was a _huge_ benefit in debugging this :D. We should keep it for debugging purposes. But, yes, we can use the Minio Client for the config.